README.md

```diff
@@ -422,6 +423,30 @@ The content features a conversation between two individuals discussing various t
 10. Nietzsche's walks
 ```
 
+## LLM Providers / LocalLLMs
+`fabric` leverages LiteLLM to enable users to seamlessly utilize a wide array of LLM Providers and Models. With LiteLLM, users can effortlessly switch between different providers and models by specifying the desired provider and model name through command line options. `fabric` utilizes LiteLLM to dynamically load the specified provider and model, empowering users to harness the unique capabilities offered by each LLM provider or model.
+
+> Please ensure that you have configured the appropriate environment variables. For instance, set `HUGGINGFACE_API_KEY` if you intend to utilize the Hugging Face API.
+
+### Usage:
+
+To specify the provider and model when running `fabric`, use the `--model` option followed by the provider and model name.
```

Comment:
How does a user list available models to select? Perhaps I need to read on LiteLLM, but I'd expect there'd be a `--list` option of some sort for models.

Reply:
Never mind, I noticed there's a `--listmodels` option. Considering this closed.

Reply:
Right, problem is... `--listmodels` only lists the available OpenAI models at this point. We have to implement some kind of helper functionality that differentiate between the configured providers.

Reply:
I'd understand this is a required item to ship this new feature. Otherwise we need to at least let people know which models are available for them to select since these are expected to be provided as text. Is the list the one provided at https://docs.litellm.ai/docs/providers?

Reply:
The link is correct. In the beginning we can lead people to that list. In the long run I'd prefer having an internal notification for that. 
As far as I can see, there's no proprietary way to do so. LiteLLM has its internal variables of providers but cannot deliver a list of models. What models are available to which provider depends on many factors. Such as Key Restriction or simply what Ollama models were locally downloaded.
So I guess we can try to enhance the user experience as best as we can. But we won't make it right for everyone  😅